src/plonk.rs

```diff
@@ -86,6 +80,7 @@ impl<C: HaloCurve> Circuit<C> {
     pub fn generate_proof<InnerC: HaloCurve<BaseField = C::ScalarField>>(
         &self,
         witness: Witness<C::ScalarField>,
+        blinding_commitments: bool,
```

Comment:
The verifier should also receive this flag right? If so I think it may be best as a `CircuitBuilder` config method (like `builder.set_blinding(true/false)`), so the user configures it only in one place

Reply:
Hm why would the verifier need to know if the commitments are blinded?

Reply:
Oh I was imagining if blinding was off, we could skip that Schnorr-based protocol in Halo (and just have the prover send `a'_0` directly), but maybe that's not significant and not worth optimizing.

How about on the prover's side -- would the benefit of disabling blinding commitments be saving some `[r] H` multiplications, or is there more to it?

I think the way you have it seems reasonable actually, just want to make sure I understand the intention.

Reply:
Yes without blinding we can skip the Schnorr protocol and spare a few operations. 
The way I see it, we would turn off blinding only for inner proofs in a recursive argument, where only the last step needs blinding. In particular, the out-of-circuit verifier would practically never be run on a proof without blinding.
The in-circuit verifier however should practically be run only on proof without blinding. So I could remove the Schnorr protocol from `recursive_verification_circuit` in this case.

Reply:
Hm I think if we consider use cases outside of Mir, there could be potential uses for all the modes -- like if someone was (out-of-circuit) verifying a SNARK-based VDF, that could be non-blinded since there's no privacy requirement, or if a block producer was recursing on the previous block proof, that might be blinded if the inner proof was generated by a different party. So I think ideally we should support all modes, but it's probably a minor cost savings so feel free to leave it for later.